include/benchmark/benchmark_api.h

```diff
@@ -331,16 +344,19 @@ class State {
   // Returns true if the benchmark should continue through another iteration.
   // NOTE: A benchmark may not return from the test until KeepRunning() has
   // returned false.
-  bool KeepRunning() {
-    if (BENCHMARK_BUILTIN_EXPECT(!started_, false)) {
-      StartKeepRunning();
-    }
-    bool const res = total_iterations_++ < max_iterations;
-    if (BENCHMARK_BUILTIN_EXPECT(!res, false)) {
-      FinishKeepRunning();
-    }
-    return res;
-  }
+  bool KeepRunning();
+
+  // Returns true if the benchmark should process another 'n' iterations. The
```

Comment:
This seems to explicitly allow the batch size to change within the same run of a benchmark, ie the argument passed to the first call differs from the argument passed to the second (within the same run). Is this behavior intentional? Is it needed?

It appears to me that if the batch size is fixed we can choose a `max_iteration` count that is exactly divisible by `kBatchSize`. Not sure how much utility that will have, but I just wanted to inquire.